src/Infrastructure.EntityFramework/Repositories/UserRepository.cs

```diff
@@ -81,6 +81,137 @@ public async Task<ICollection<Core.Entities.User>> SearchAsync(string email, int
         }
     }
 
+    public async Task<ICollection<DataModel.UserDetails>> SearchDetailsAsync(string email, int skip, int take)
+    {
+        using (var scope = ServiceScopeFactory.CreateScope())
+        {
+            var dbContext = GetDatabaseContext(scope);
+            List<DataModel.UserDetails> users;
+
+            if (dbContext.Database.IsNpgsql())
+            {
+                users = await GetDbSet(dbContext)
+                    .Where(e => email == null || EF.Functions.ILike(EF.Functions.Collate(e.Email, "default"), $"{email}%"))
+                    .OrderBy(e => e.Email)
+                    .Skip(skip).Take(take)
+                    .Select(e => new DataModel.UserDetails
```

Comment:
question: can we move all this init logic to the model itself?

Reply:
Yes! Good call.